### ISSUE TITLE
Update CI config to test multiple Linux Distributions

### ISSUE THREAD

**Issue by bkochuna** [bkochuna/ners570f20-Lab06#39]
This ticket is about updating the `.travis.yml` configuration so that this project is tested for multiple linux distributions.

To complete this ticket the `.travis.yml` should be updated so that the test script(s) is executed for the following linux distributions supported on Travis-CI
1. Trusty
2. Xenial (default)
3. Bionic
4. Focal

The Precise linux distrubtion of Ubuntu is also available but support ended for this in 2017 and the default CMake and g++ compiler are too old.

[Full List of Ubuntu Releases](https://wiki.ubuntu.com/Releases)

**Comment by knbarton** [comment]
I assigned myself to this if that's okay.

**Comment by bkochuna** [comment]
Definitely! Good luck. Since this is EC, we won't give hints. I did figure this one out, it took a little longer than I expected because I tried to do it for a more complex workflow which led me to discovering that the distros are an exception to job matrixes and work differently than every other one.

I also tried to get it working from Trusty, but this distribution is too old. The default g++ compiler does not support C++11, so it would require installing a newer compiler (which was too much to ask).
